### PR TITLE
Ticket40 link users to commitments and criteria

### PR DESCRIPTION
**Pull request by sergiomarrocoli**
hard to test right now, but you can run seeds, go to dashboard and try to delete/edit the commitments. if the cmmitment is yours it will destro/go to edit page. if not, it won't

### CODE REVIEW

File: app/controllers/commitments_controller.rb [hunk 1]
```diff
@@ -86,6 +93,11 @@ def destroy
 
   private
 
+  def criterium_id_valid?
+    criterium = Criterium.find_by(id: params[:criterium_id])
+    criterium && criterium.commitment.nil? && criterium.user_id && criterium.user_id == current_user.id
```
Comment (yndajas) [inline app/controllers/commitments_controller.rb:98]: Is it worth using a `try` or two here to make it a little more concise? E.g. `criterium.try(:user_id) == current_user.id` Not sure if there's a way to get the first half `try` or `&.`-y given the use of `nil?`

File: test/controllers/commitments_controller_test.rb [hunk 2]
```diff
@@ -21,10 +21,23 @@ class CommitmentsControllerTest < ActionDispatch::IntegrationTest
...
     assert_response :success
   end
 
+  test "should not GET new without a criterium_id" do
+    sign_in users(:user_1)
+    get new_commitment_url
+    assert_redirected_to new_criterium_path
+  end
+
+  test "should not GET new if the criterium_id matches a different criterium" do
```
Comment (yndajas) [inline test/controllers/commitments_controller_test.rb:34]: Would this be clearer as something like "should not GET new if the criterium_id is already associated with a commitment"?

**Review by yndajas — CHANGES_REQUESTED** [review]
I seem to be able to delete anything from the dashboard. E.g. after seeding and importing, the second commitment belongs to user 2, but if I'm logged in as user 1 I can still delete it

Other than that, the tests pass, and just a couple of suggestions in the comments

File: test/controllers/commitments_controller_test.rb (hunk 2 above)
Comment (sergiomarrocoli) [reply test/controllers/commitments_controller_test.rb]: changed

**Comment by sergiomarrocoli** [comment]
> I seem to be able to delete anything from the dashboard. E.g. after seeding and importing, the second commitment belongs to user 2, but if I'm logged in as user 1 I can still delete it
> 
> Other than that, the tests pass, and just a couple of suggestions in the comments

Made the changes requested. Can you try resetting the database, server etc? i'm prevented from deleting/editing anything not linked to my user, and the test that check for that also pass, so think it's probably something to with that

**Review by yndajas — APPROVED** [review]
Thought I reset everything before, but it's working now anyway!

**Pull request merged**